src/cargo/util/graph.rs

```diff
@@ -1,35 +1,36 @@
 use std::borrow::Borrow;
-use std::collections::{HashMap, HashSet};
+use std::collections::BTreeSet;
 use std::fmt;
-use std::hash::Hash;
 
-pub struct Graph<N, E> {
-    nodes: HashMap<N, HashMap<N, E>>,
+use im_rc;
+
+pub struct Graph<N: Clone, E: Clone> {
+    nodes: im_rc::OrdMap<N, im_rc::OrdMap<N, E>>,
```

Comment:
aha, I see how this would make things a bit speedier on the cloning half of things

Reply:
(maybe `RcList` is now obsoleted?)